Studio-Lovelies/GG-JointJustice-Unity
# Player can access the profile menu

## Summary

Updates the evidence menu to support showing profiles. Actor dictionaries now use the same system as evidence (dictionary created on a monobehaviour instead of in the scriptable object). The logic from EvidenceDictionary is now in its own class called ObjectDictionary which the ActorDictionary and EvidenceDictionary monobehaviours use. ActorData and Evidence now implement ICourtRecordObject which allows their names, icons, and descriptions to be accessed. Also features tests for the evidence menu and ObjectDictionary class.

## Before/after screenshots and/or animated gif

![profilesdemo2](https://user-images.githubusercontent.com/67964179/132934810-b9e0b182-6462-4ef6-8edf-f42d955261bf.gif)

## Testing instructions

Open a scene with the StandardInclusion asset and enter play mode. Press z to open the evidence menu, then press c to open the profiles menu. Navigate with the arrow keys (and check for spelling mistakes in each bio).
Open the unity test runner and run the play mode and edit mode tests.
The screenshots below show the tests passing.
<img width="564" alt="Screenshot 2021-09-16 at 14 30 48" src="https://user-images.githubusercontent.com/67964179/133621427-878c360c-a616-4be8-abfb-f081994640df.png">
<img width="565" alt="Screenshot 2021-09-16 at 14 30 55" src="https://user-images.githubusercontent.com/67964179/133621433-399d8f7d-b0a6-4863-823f-9e8268b1f952.png">

## Additional information

- `[x]` Changes UI
- `[x]` Introduces new feature
- `[ ]` Removes existing feature
- `[x]` Has associated resource: 
  
#11
